### ISSUE TITLE
svelte/valid-prop-names-in-kit-pages marks Snippet props as errors

### ISSUE THREAD

**Issue by VsevolodGolovanov** [sveltejs/eslint-plugin-svelte#1199]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.24.0

### What version of `eslint-plugin-svelte` are you using?

3.5.1

### What did you do?

<details>
<summary>Configuration</summary>
I only changed `no-unused-vars` to `warn` from the default starter config (project created by following official instructions).

```
export default ts.config(
	includeIgnoreFile(gitignorePath),
	js.configs.recommended,
	...ts.configs.recommended,
	...svelte.configs.recommended,
	prettier,
	...svelte.configs.prettier,
	{
		languageOptions: {
			globals: {
				...globals.browser,
				...globals.node
			}
		}
	},
	{
		files: ['**/*.svelte', '**/*.svelte.ts', '**/*.svelte.js'],
		ignores: ['eslint.config.js', 'svelte.config.js'],

		languageOptions: {
			parserOptions: {
				projectService: true,
				extraFileExtensions: ['.svelte'],
				parser: ts.parser,
				svelteConfig
			}
		}
	},
	{
		rules: {
			'@typescript-eslint/no-unused-vars': 'warn'
		}
	}
);
```
</details>

The snippet is typed like `https://svelte.dev/docs/svelte/snippet#Typing-snippets` demonstrates:
```svelte
interface Props {
  children: Snippet;
  layoutHeaderCenter: Snippet;
}

const { layoutHeaderCenter, children }: Props = $props();
```

### What did you expect to happen?

I expected no error.

### What actually happened?

I got an ESLint error:

![Image](https://github.com/user-attachments/assets/590b823b-3247-46dc-ad60-15f55b0e4e51)

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/VsevolodGolovanov/eslint-plugin-svelte-snippet-error-reproducer

**Comment by VsevolodGolovanov** [comment]
Wait, this doesn't work actually - I'm defining this `layoutHeaderCenter` snippet in the subroute page, and it doesn't render.
It seems like custom snippets don't work in layouts: https://github.com/sveltejs/svelte/discussions/14108
That's unexpected...

**Comment by baseballyama** [comment]
@VsevolodGolovanov So, that means the behavior of `eslint-plugin-svelte` is correct for now, right?

**Comment by VsevolodGolovanov** [comment]
@baseballyama, yes, looks like it.
